docs/user_guide.md

```diff
@@ -528,14 +527,38 @@ This section is only applicable to advanced users who wish to create custom
 constraint templates. If the existing templates are sufficient, you can skip
 this section.
 
-#### Template Naming Convention
+#### Template Authoring Convention
+
+##### Naming
+
+The template name appears in three places in a template YAML file:
+
+*   metadata name: All lower case with "-" as the separator. It has the format
+    of "gcp-{resource}-{feature}-{version}" (example: "gcp-storage-logging-v1").
```

Comment:
Do we think the "gcp" prefix is needed here? It feels redundant to append a "gcp" prefix in front of names like "gce", "gke", or "cloudsql". 

We could adopt a convention where the "gcp" prefix is used if the constraint applies to multiple resources. Thoughts?

Reply:
I do feel like gcp might be a bit redundant here, unless we plan to go with multiple cloud in the future then this would make sense, is that part of the plan?

Reply:
That's a possibility. In that case, having a "gcp" prefix sounds preferable. 

But we still have this redundancy issue. One option is not use terms like "gce", "gcs", or "gke".
Instead, we'll use "gcp-compute", "gcp-storage", and "gcp-kubernetes" (or should it be "gcp-container-clusters" if we want to follow gcloud's convention?)

Thoughts?

Reply:
i think having the `gcp` prefix across the board is helpful to differentiate from, at a minimum, Kubernetes constraints.

In terms of redundancy, I think `gcp-compute` and such (following gcloud conventions) make sense.

Reply:
Do you guys think the file name might get too long if we put too much info there?

Reply:
I don't think that's a major issue, but I also don't think this is actually an area where either decision has a big impact.